### PR TITLE
Bump rewrite to 8.12.0

### PR DESCRIPTION
**Pull request by fabapp2**
- Remove some custom Recipes with Rewrite code
- Fix some test data to use available version
- Add some TODOs and FIXMEs

### CODE REVIEW

File: src/main/java/org/springframework/cli/merger/ProjectMerger.java [hunk 1]
```diff
@@ -446,11 +429,35 @@ private void mergeMavenDependencyManagement(Path currentProjectPomPath, Model mo
...
+				List<SourceFile> pomFiles = mavenParser.parse(paths, this.currentProjectPath, getExecutionContext()).toList();
+//				String version = dependency.getVersion();
+//				if(dependency.getVersion().contains("${")) {
+//					version = pomFiles.get(0).getMarkers().findFirst(MavenResolutionResult.class).get().getPom().getProperties().
+//				}
+
 
-				List<? extends SourceFile> pomFiles = mavenParser.parse(paths, this.currentProjectPath, getExecutionContext());
-				List<Result> resultList = addManagedDependency.run(pomFiles).getResults();
+				Recipe addManagedDependency = new Recipe() {
```
Comment (fabapp2) [inline src/main/java/org/springframework/cli/merger/ProjectMerger.java:439]: Move into factory

File: src/main/java/org/springframework/cli/merger/ProjectMergerWithMavenDepRecipe.java [hunk 2]
```diff
@@ -72,6 +54,19 @@ public ProjectMergerWithMavenDepRecipe(Path pathToMerge, Path pathCurrentProject
...
+
+	/**
+	 * this approach does not work in our use case since it needs to construct a classpath for the application, that is the 'spring up shell'.
+	 * The rationale is that the recpie does not want to add a dependency, in particular in a multi-module project, to module that does not
+	 * use the code.  This involves loading classes from the classpath.  fall back to simple XML manipulation.
+	 */
+
+
+	/* FIXME: Code has no usages
 	public void merge() throws IOException {
```
Comment (fabapp2) [inline src/main/java/org/springframework/cli/merger/ProjectMergerWithMavenDepRecipe.java:70]: Delete unused code

File: src/main/java/org/springframework/cli/merger/ProjectMergerWithMavenDepRecipe.java [hunk 3]
```diff
@@ -114,32 +109,26 @@ public void merge() throws IOException {
...
-		 */
-
-
 
 		List<Dependency> dependenciesToMerge = modelToMerge.getDependencies();
 
 		// Need to get source files
-		// TODO detect if java 11 or 8 or whatever....? JavaParser.fromJavaVersion() ?
-		JavaParser javaParser = new Java17Parser.Builder()
+		JavaParser javaParser = JavaParser.fromJavaVersion()
```
Comment (fabapp2) [inline src/main/java/org/springframework/cli/merger/ProjectMergerWithMavenDepRecipe.java:124]: @jvalkeal Does changing to dynamic lookup of JavaParser impact success of cretaing native app?

File: src/main/java/org/springframework/cli/recipe/AddDependencyRecipeFactory.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ */
+public class AddDependencyRecipeFactory {
+
+    /**
+     * Create {@link AddDependency} recipe from Maven dependency XML snippet.
+     */
+    public Recipe create(String mavenDependency) {
+        try {
+            XmlMapper mapper = new XmlMapper();
+            JsonNode jsonNode = null;
```
Comment (fabapp2) [inline src/main/java/org/springframework/cli/recipe/AddDependencyRecipeFactory.java:42]: rename this to xmlNode

File: src/main/java/org/springframework/cli/recipe/InjectTextMavenRepositoryRecipe.java
Comment (fabapp2) [inline src/main/java/org/springframework/cli/recipe/InjectTextMavenRepositoryRecipe.java:1]: Use factory approach here

File: src/main/java/org/springframework/cli/merger/ProjectMergerWithMavenDepRecipe.java (hunk 3 above)
Comment (jvalkeal) [reply src/main/java/org/springframework/cli/merger/ProjectMergerWithMavenDepRecipe.java:124]: dunno, we'll see when at some point graal build get more stable.

**Comment by jvalkeal** [comment]
lgtm. btw you should not PR from main branch.

**Pull request merged**